### PR TITLE
DEPR: deprecate Index.is_boolean

### PR DESCRIPTION
**Pull request by topper-123**
- [ ] progress towards #50042
- [x] [Tests added and passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#writing-tests) if fixing a bug or adding a new feature
- [x] All [code checks passed](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#pre-commit).
- [ ] Added [type annotations](https://pandas.pydata.org/pandas-docs/dev/development/contributing_codebase.html#type-hints) to new arguments/methods/functions.
- [x] Added an entry in the latest `doc/source/whatsnew/vX.X.X.rst` file if fixing a bug or adding a new feature.

### CODE REVIEW

**Review by phofl — COMMENTED** [review]
Can you add tests as well? Same as the other, have to hold off a bit

**Comment by topper-123** [comment]
> Can you add tests as well?

I added a test in pandas/tests/indexes/common.py. Do you mean something additional?

**Comment by phofl** [comment]
Missed this, sorry

**Comment by phofl** [comment]
needs a rebase, can merge afterwards

File: pandas/core/indexes/base.py [hunk 1]
```diff
@@ -2210,6 +2213,12 @@ def is_boolean(self) -> bool:
         >>> idx.is_boolean()
         False
         """
+        warnings.warn(
+            f"{type(self).__name__}.is_boolean is deprecated."
```
Comment (phofl) [inline pandas/core/indexes/base.py:2217]: Same as the other pr, needs a whitespace here

**Comment by topper-123** [comment]
Updated.

**Review by phofl — APPROVED** [review]

**Comment by phofl** [comment]
Thx @topper-123

**Pull request merged**